core/src/main/java/com/exadel/aem/toolkit/core/injectors/EToolboxListInjector.java

```diff
@@ -0,0 +1,174 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exadel.aem.toolkit.core.injectors;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Array;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.StreamSupport;
+import javax.annotation.Nonnull;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.models.spi.DisposalCallbackRegistry;
+import org.apache.sling.models.spi.Injector;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+import com.exadel.aem.toolkit.api.annotations.injectors.EToolboxList;
+import com.exadel.aem.toolkit.core.lists.utils.ListHelper;
+
+/**
+ * Injects EToolbox Lists into Sling models obtained via {@code ResourceResolver} instance
+ * @see ListHelper
+ * @see Injector
+ */
+@Component(service = Injector.class,
+    property = Constants.SERVICE_RANKING + ":Integer=" + InjectorConstants.SERVICE_RANKING
+)
+public class EToolboxListInjector implements Injector {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EToolboxListInjector.class);
+
+    public static final String NAME = "eak-etoolbox-list-injector";
+
+    /**
+     * Retrieves the name of the current instance
+     * @return String value
+     * @see Injector
+     */
+    @Nonnull
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    /**
+     * Attempts to inject list entries into the given adaptable
+     * @param adaptable        A {@link SlingHttpServletRequest} or a {@link Resource} instance
+     * @param name             Name of the Java class member to inject the value into
+     * @param type             Type of receiving Java class member
+     * @param element          {@link AnnotatedElement} instance that facades the Java class member allowing to retrieve
+     *                         annotation objects
+     * @param callbackRegistry {@link DisposalCallbackRegistry} object
+     * @return The value to inject, or null in case injection is not possible
+     * @see Injector
+     * @see ListHelper
+     */
+    @Override
+    public Object getValue(
```

Comment:
I would suggest splitting this method, as it is pretty long already and contains lots of if-else statements. 
Perhaps some of these "branches" of logic can be extracted into a separate method?

Reply:
I edited the JavaDoc and moved some of the logic from the getValue() method into separate methods.